scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java

```diff
@@ -192,9 +192,17 @@ private boolean runningTaskMustBeResetToPending(InternalTask task, RMProxy rmPro
 
             if (taskNodesKnownByRM) {
                 TaskLauncher launcher = task.getExecuterInformation().getLauncher();
-                logger.info("Recover running task " + task.getId() + " (" + task.getName() +
-                            ") successfully with task launcher " + launcher);
-                resetToPending = false;
+                logger.debug("Checking whether task launcher has called its doTask method: " +
+                             launcher.getDoTaskInvoked());
+                if (launcher.getDoTaskInvoked()) {
+                    logger.info("Recover running task " + task.getId() + " (" + task.getName() +
+                                ") successfully with task launcher " + launcher);
+                    resetToPending = false;
+                } else {
+                    logger.info(FAIL_TO_RECOVER_RUNNING_TASK_STRING + task.getId() + " (" + task.getName() +
+                                ") because its task launcher has not started to execute the task");
+                    resetToPending = true;
```

Comment:
I suppose we need to do something with the task launcher. Unless the rm takes care of (kill) it when we release the node.

Reply:
Ok, makes sense to initiate a kill here. But after looking at it, it seems that TaskLauncher#kill is not what we want. I would instead make TaskLauncher#terminate public and call it to stop the task launcher activity. How does it sound?

Reply:
I think the nodes must be released to the RM, they will be busy in that case (and stay so if nothing is done)

Reply:
ok, after the recovery, there is a mechanism that releases the busy nodes that have no running task. What you describe falls in this specific case, so there is no need to add a killing mechanism at this point.